engines/gob/goblin_v7.cpp

```diff
@@ -0,0 +1,584 @@
+			return direction;
+	}
+	return 0;
+}
+
+bool Goblin_v7::directionWalkable(int8 x, int8 y, int8 direction) {
+	int8 nextX = x + deltaXFromDirection[direction];
+	int8 nextY = y + deltaYFromDirection[direction];
+	if (nextX >= 0 &&
+		nextX < _vm->_map->getMapWidth() &&
+		nextY >= 0 &&
+		nextY < _vm->_map->getMapHeight()) {
+		return positionWalkable(_vm->_map, nextX, nextY);
+	}
+	else
+		return false;
+}
+
+int32 Goblin_v7::bestWalkableDirectionFromOriginAndDest(int8 x, int8 y, int8 destX, int8 destY) {
+	int8 deltaX = 0;
+	int8 deltaY = 0;
+
+	if (destX < x)
+		deltaX = -1;
+	else if (destX > x)
+		deltaX = 1;
+
+	if (destY < y)
+		deltaY = -1;
+	else if (destY > y)
+		deltaY = 1;
+
+	int8 direction = directionFromDeltaXY(deltaX, deltaY);
+	if (directionWalkable(x, y, direction))
+		return direction;
+
+	// Look for another walkable direction
+	direction -= 1;
+	if (direction <= 0)
+		direction += 8;
+	if (directionWalkable(x, y, direction))
+		return direction;
+
+	direction += 2;
+	if (direction > 8)
+		direction -= 8;
+	if (directionWalkable(x, y, direction))
+		return direction;
+
+	direction -= 3;
+	if (direction <= 0)
+		direction += 8;
+	if (directionWalkable(x, y, direction))
+		return -direction;
+
+	direction += 4;
+	if (direction > 8)
+		direction -= 8;
+	if (directionWalkable(x, y, direction))
+		return -direction;
+
+	return 0;
+}
+
+
+int32 Goblin_v7::findPath(int8 x, int8 y, int8 destX, int8 destY) {
+	int8 currentX = x;
+	int8 currentY = y;
+	int8 returnToPreviousStepDir = -1;
+	int8 var_8 = 0;
+	int8 firstDirection = 0;
+	int8 var_1C = 0;
+
+	while (true)
+	{
+		int8 currentDirection = bestWalkableDirectionFromOriginAndDest(currentX, currentY, destX, destY);
+		if (currentDirection == 0)
+			return 0;
+
+		if (currentDirection >= 0) {
+			if (var_8 == 1) {
+				var_8 = 2;
+				var_1C = findPath(x, y, currentX, currentY);
+				if (var_1C > 0)
+					firstDirection = var_1C;
+			}
+		}
+		else {
+			currentDirection = -currentDirection;
+			if (var_8 == 0)
+				var_8 = 1;
+		}
+
+		if (returnToPreviousStepDir > 0) {
+			returnToPreviousStepDir += 4;
+			if (returnToPreviousStepDir > 8)
+				returnToPreviousStepDir -= 8;
+		}
+
+		if (currentDirection == returnToPreviousStepDir) {
+			currentDirection += 4;
+			if (currentDirection > 8)
+				currentDirection -= 8;
+
+			if (!directionWalkable(currentX, currentY, currentDirection))
+				return 0;
+		}
+
+		if (firstDirection == 0)
+			firstDirection = currentDirection;
+
+		returnToPreviousStepDir = currentDirection; // Will be inverted later
+		currentX += deltaXFromDirection[currentDirection];
+		currentY += deltaYFromDirection[currentDirection];
+
+		if (currentX != destX || currentY != destY)
+			continue;
+		else
+			return firstDirection;
+	}
+}
+
+int32 Goblin_v7::computeObjNextDirection(Mult::Mult_Object &obj) {
+	Mult::Mult_AnimData animData = *obj.pAnimData;
+	if (animData.stateType == 1) {
+		warning("Adibou2 computeObjNextDirection animData.stateType == 1 stub");
```

Comment:
Should it really be called `Adibou2` here?

Reply:
No, indeed, I will change to "Goblin_v7::computeObjNextDirection".